### PR TITLE
Reuse existing CKShare if it already exists

### LINKED ISSUE #5: Share is being replaced
Every time the user taps the Share action on a contact row a new CKShare is created with the contact record id.

This means that any existing shares are revoked, and anyone you've shared a contact with before will lose the shared contact. It also means that we never get to see the awesome view in `UICloudSharingController` where one can edit who currently has access and invite more users.

This view looks like this 👇🏽

Screen Shot 2021-06-11 at 20 13 35

### PR DESCRIPTION
**Pull request by JagCesar**
If a contact is already shared we use the existing CKShare object.

fixes #5

### CODE REVIEW

File: Sharing/ViewModels/ViewModel.swift
Comment (JagCesar) [inline Sharing/ViewModels/ViewModel.swift:161]: The documentation and function name are no longer correct. If we agree that this PR solves a problem we can change this by either changing the name & documentation, or creating a separate function that calls this one. Something like this:

```
func share(… // This function checks if `contact.associatedRecord.share != nil` and based on that uses one of the two private functions below

private func createShare(…

private func getExistingShare(…
```

File: Sharing/ViewModels/ViewModel.swift
Comment (sjrmanning) [inline Sharing/ViewModels/ViewModel.swift:15]: Since we only use this in one place, how about a slightly more descriptive name here? Hopefully it's never used, but maybe something simple like:

```suggestion
        case invalidRemoteShare
```

File: Sharing/ViewModels/ViewModel.swift
Comment (sjrmanning) [inline Sharing/ViewModels/ViewModel.swift:186]:
```suggestion
                completionHandler(.failure(ViewModelError.invalidRemoteShare))
```

**Review by sjrmanning — CHANGES_REQUESTED** [review]
Thanks for your PR and for filing the issue! Apologies for the delay on this review. Just a couple small changes requested.

Just as a note before we create a fully-fledged guide to contributing, while we want to keep these samples as simple and contrived as possible (to more clearly demonstrate specific CloudKit API usage), we're approving this change as its scope is limited, it helps to demonstrate CloudKit features, and doesn't significantly increase the overall complexity of the project.

File: Sharing/ViewModels/ViewModel.swift
Comment (JagCesar) [reply Sharing/ViewModels/ViewModel.swift]: Fixed in https://github.com/apple/cloudkit-sample-sharing/pull/6/commits/afef4791f76d59a263194b3efc9c24825683165f

File: Sharing/ViewModels/ViewModel.swift
Comment (JagCesar) [reply Sharing/ViewModels/ViewModel.swift]: Fixed in https://github.com/apple/cloudkit-sample-sharing/pull/6/commits/d954223a60fc98a85f9890ae5786128b98923dc7

**Comment by JagCesar** [comment]
@sjrmanning To keep things as simple as possible I only did the requested changes. If anything more needs to be fixed before merging I've ticked the `Allow edits by maintainers`-box, so you have write access to my fork.

Thanks for taking your time to review! 🙏

**Review by sjrmanning — APPROVED** [review]
Thanks @JagCesar ! By the way in case you're interested, I've pulled over this change into the `swift-concurrency` branch in #8. Cheers!

**Pull request merged**
